lib/api_browser/app/views/action.html

```diff
@@ -4,99 +4,94 @@
-                <span ng-if="!hasResponseExample(response)">
-                  {{response.name}}
-                </span>
-              </td>
-              <td>
-                {{response.media_type.id || response.media_type.identifier}}
-              </td>
-              <td>
-                <attribute-description attribute="response"></attribute-description>
-              </td>
-            </tr>
-          </tbody>
-        </table>
+      <h2>Request Example</h2>
+      <div ng-if="!action.payload.examples">
+        <h4>URL</h4>
+        <url action="action" example></url>
       </div>
+      <tabset ng-if="action.payload.examples">
+        <tab ng-repeat="(type, example) in action.payload.examples" heading="{{type}}">
+          <h4>URL</h4>
+          <url action="action" example></url>
+          <h4>Content-Type</h4>
+          <p><code>{{example.content_type}}</code></p>
+          <h4>Request body</h4>
+          <pre>{{ example.body }}</pre>
+        </tab>
+      </tabset>
     </div>
+  </div>
 
-    <div class="col-lg-12" ng-repeat="response in responses | filter: hasResponseExample" id="action-{{action.name}}-{{response.name}}">
-      <h4>Response example: {{ response.name }}</h4>
-
-      <div class="row">
-        <div class="col-md-4">
-          <h5>Status</h5>
-          <p>{{ response.status }}</p>
-          <h5>Content-Type</h5>
-          <p>{{ response.jsonExample.content_type }}</p>
-        </div>
-        <div class="col-md-8">
-          <h5>Example</h5>
-          <pre>{{ response.jsonExample.body }}</pre>
+  <div class="row" ng-if="hasResponses()">
+    <div class="col-lg-12">
+      <h2>Responses</h2>
+      <div class="panel panel-default" ng-repeat="response in responses">
+        <div class="panel-heading"><h4 class="panel-title">{{response.name}}</h4></div>
+        <div class="panel-body">
+          <div class="row">
+            <div class="col-md-4">
+              <h5>Status</h5>
+              <p>{{ response.status }}</p>
+              <div ng-if="response.examples.json">
+                <h5>Content-Type</h5>
+                <p>{{ response.examples.json.content_type }}</p>
+              </div>
+              <div ng-if="response.payload.id">
+                <h5>Media Type</h5>
+                <p><a ui-sref="root.type({version: apiVersion, type: response.payload.id})">{{response.payload.name}}</a></p>
+              </div>
+            </div>
+            <div class="col-md-8">
+              <div ng-if="response.payload.examples">
+                <h5>Response Example</h5>
+                <tabset ng-if="numExamples > 1">
+                  <tab ng-repeat="(type, example) in response.payload.examples" heading="{{type}}">
+                    <pre>{{ example.body }}</pre>
+                  </tab>
+                </tabset>
+                <div ng-if="response.numExamples == 1">
+                  <pre ng-repeat="(type, example) in response.payload.examples">{{ example.body }}</pre>
```

Comment:
I don't think the type is too helpful if there is only one (that's why the special case exists). The repeat is there only as a convenient way how to get to the `example` object.